tokio/src/sync/broadcast.rs

```diff
@@ -901,12 +902,14 @@ impl<T> Shared<T> {
                         // Safety: `tail` lock is still held.
                         let waiter = unsafe { waiter.as_mut() };
```

Comment:
You can no longer take a mutable reference to the entire waiter. You don't have mutable access to `queued` anymore.

Reply:
Good catch.